### DISCUSSION TITLE
Problem with deleting a referenced collection's ObjectId from the main collection

### DISCUSSION THREAD

**Discussion by AlexDernov** [neuefische/questions#679] [Web] [answered]
**I have** two collections "locations" and "notes" in my database, connected via _const location = await Location.findById(id).populate("notes");_

**Problem:** Deleting a note from a notes list deletes the note from the corresponding collection in the database, but does not remove its ObjectId from the Array "notes" in the "locations" collection. 

**In components/Note/index.js**
const responseNote = await fetch(`/api/notes/${note?._id}`, {
      method: "DELETE", ... **is OK!**
**BUT**
const responseLocation = await fetch(`/api/locations/${locatData?._id}`, {
        method: "PATCH",
        headers: {
          { "Content-Type": "application/json",
        },
        body: JSON.stringify({
          notes: [
            locatData?.notes.filter((oneNote) => {
            console.log("oneNote", oneNote);
            console.log("note._id", note._id);
             return oneNote._id == note._id ? false : true}...
**is NOT OK!**

A google search and attempts to rewrite the code according to the suggested solutions were unsuccessful.

[My actual PR](https://github.com/AlexDernov/capestone-project-hungryOar/pull/11)

<img width="726" alt="Screenshot 18 08 2023 um 13 11 05 PM" src="https://github.com/neuefische/questions/assets/127208943/9bae667f-e75b-4ab5-b3ac-d9444d906624">

Еhanks in advance for your help!

**Comment by F-Kirchhoff** [comment]
I left a comment on your PR where you need to change your code. 

You need to manually remove the noteID from the id-array inside the corresponding location document. I added some hints how you can do this. 

An alternative would be to reverse the linking between location and note. If you store the location id inside the note as a reference and you remove the noteId Array in the locations, the linking is deleted alongside with the note. Of course you would need to adapt your code that all notes are filtered for the one with the correct location when fetching the location data (you couldn't use `.populate` anymore). Both ways are valid and have their ups and down. Its up to you how you wanna implement it.

**Reply by AlexDernov** [reply]
Thank you!!!

**Comment by Kheirah** [comment, marked as answer]
Found it! 😆 In the `index` of Note:
```js
notes: [ //the outer array is redundant as the filter function already returns an array.
  locatData?.notes.filter((oneNote) => {
    return oneNote._id == note._id ? false : true}
  ),
],
```
Corrected:
```js
notes: locatData?.notes.filter((oneNote) => {
    return oneNote._id == note._id ? false : true}
  ),
```

**Reply by AlexDernov** [reply]
Thank you very much!!
